recipes/civetweb/all/conanfile.py

```diff
@@ -39,22 +45,45 @@ def _build_subfolder(self):
         return "build_subfolder"
 
     def config_options(self):
-        if self.settings.os == 'Windows':
+        if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
         if not self.options.with_cxx:
+            del self.settings.compiler.cppstd
             del self.settings.compiler.libcxx
+        if not self.options.with_ssl:
+            del self.options.ssl_dynamic_loading
 
     def requirements(self):
         if self.options.with_ssl:
             self.requires("openssl/1.1.1i")
 
+    def validate(self):
+        if self.options.get_safe("ssl_dynamic_loading") and not self.options["openssl"].shared:
+            raise ConanInvalidConfiguration("ssl_dynamic_loading requires shared openssl")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("civetweb-%s" % self.version, self._source_subfolder)
+
+    def _patch_sources(self):
+        cmakelists_src = os.path.join(self._source_subfolder, "src", "CMakeLists.txt")
+        tools.replace_in_file(cmakelists_src, "${OPENSSL_INCLUDE_DIR}", "${OpenSSL_INCLUDE_DIR}")
+        tools.replace_in_file(cmakelists_src, "${OPENSSL_LIBRARIES}", "OpenSSL::SSL OpenSSL::Crypto")
```

Comment:
If you link with the CMake targets only one is required

```suggestion
        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "CMakeLists.txt"),
             "${OPENSSL_LIBRARIES}", "OpenSSL::SSL OpenSSL::Crypto")
```

Reply:
Yes but `${OPENSSL_INCLUDE_DIR}` is also displayed in a message, it avoids a confusing message at configure time.